types/message.go

```diff
@@ -0,0 +1,97 @@
+package types
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/uuid"
+	"go.vocdoni.io/dvote/log"
+)
+
+// Message is the JSON API body message used by the CSP and the client
+type Message struct {
+	Error     string       `json:"error,omitempty"`
+	TokenR    HexBytes     `json:"token,omitempty"`
+	AuthToken *uuid.UUID   `json:"authToken,omitempty"`
+	Payload   HexBytes     `json:"payload,omitempty"`
+	Signature HexBytes     `json:"signature,omitempty"`
+	SharedKey HexBytes     `json:"sharedkey,omitempty"`
+	Title     string       `json:"title,omitempty"`     // reserved for the info handler
+	AuthType  string       `json:"authType,omitempty"`  // reserved for the info handler
+	AuthSteps []*AuthField `json:"authSteps,omitempty"` // reserved for the info handler
+	AuthData  []string     `json:"authData,omitempty"`  // reserved for the auth handler
+	Response  []string     `json:"response,omitempty"`  // reserved for the handlers
+	Elections []*HexBytes  `json:"elections,omitempty"` // reserved for the indexer handler
```

Comment:
nit: you probably want `[]HexBytes`, because each one of those is just a slice, which is pretty small already. Adding a pointer doesn't help, similar to how `[]*[]byte` isn't better than `[][]byte`.